src/commands/run.ts

```diff
@@ -405,18 +406,56 @@ const fluenceRun = async (args: RunArgs) => {
   }
 
   await initFluenceClient(args, args.maybeFluenceConfig);
-  const { Fluence, callAquaFunction } = await import("@fluencelabs/js-client");
+
+  const { Fluence, callAquaFunction, js2aqua, aqua2js } = await import(
+    "@fluencelabs/js-client"
+  );
+
+  const schema = functionCall.funcDef;
+
+  // TODO: remove this after DXJ-535 is done
+  const validatedRunData = Object.fromEntries(
+    Object.entries(args.runData ?? {}).map(([argName, argValue]) => {
+      assert(
+        typeof argValue !== "function",
+        "Should be impossible to pass function as an argument to fluence run",
+      );
+
+      const fields =
+        schema.arrow.domain.tag === "nil" ? {} : schema.arrow.domain.fields;
+
+      const argSchema = fields[argName];
+
+      assert(
+        argSchema !== undefined,
+        "Should be impossible because schema always contains same keys as runData",
+      );
```

Comment:
but if schema.arrow.domain.tag === "nil" then fields would be empty object and this would be undefined
when schema.arrow.domain.tag === "nil" is nil?

Reply:
If `schema.domain.tag` would equal "nil", then this `map` functions wouldn't even execute a single time.

Reply:
map would execute if person puts anything in runData, user can accidentally do that easilly